src/components/Radio/index.tsx

```diff
@@ -0,0 +1,166 @@
+import styled from '@emotion/styled'
+import React, { forwardRef, InputHTMLAttributes } from 'react'
+import { color, device, fontSize, radius, space, transition } from '../..'
+import { Checkmark } from '../../icons'
+import { Label } from '../Label'
+
+const Container = styled.span`
+  position: relative;
+  display: inline-grid;
+  align-items: center;
+  justify-items: center;
+`
+
+const Icon = styled(Checkmark)`
+  grid-area: 1 / 1;
+  color: ${color.nova};
+  pointer-events: none;
+`
+
+const StyledInput = styled.input`
+  grid-area: 1 / 1;
+  background-color: ${color.nova};
+  height: 24px;
+  width: 24px;
+  appearance: none;
+  border: 1px solid ${color.spaceLightest};
+  border-radius: 50%;
+  transition: all ${transition};
+
+  &:checked {
+    background-color: ${color.earth};
+    border-color: ${color.earth};
+  }
+
+  &:disabled {
+    background-color: ${color.stardust};
+    border-color: ${color.spaceLightest};
+
+    & + ${Icon} {
+      color: ${color.stardust};
+    }
+  }
+
+  &:checked&:disabled + ${Icon} {
+    color: ${color.spaceLighter};
+  }
+
+  &:focus,
+  &:active {
+    outline: 0;
+    box-shadow: 0 0 0 ${space[4]} ${color.earthLightestAlpha};
+  }
+`
+
+type InputAttributes = Omit<InputHTMLAttributes<HTMLInputElement>, 'id'>
+
+export interface RadioProps extends InputAttributes {
+  ref?: React.Ref<HTMLInputElement>
+  id: string
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
+}
+
+export const Radio: React.FC<RadioProps> = forwardRef(
+  ({ onChange, value, checked, id, name, disabled = false }, ref) => {
+    return (
+      <Container>
+        <StyledInput
+          type="radio"
+          ref={ref}
+          onChange={onChange}
+          value={value}
+          checked={checked}
+          id={id}
+          name={name}
+          disabled={disabled}
+        />
+        <Icon size={16} />
+      </Container>
+    )
+  }
+)
+
+const Wrapper = styled.span`
+  label {
+    display: grid;
+    grid-template-columns: 56px 1fr;
+    padding: ${space[16]} ${space[16]} ${space[16]} 0;
+    align-items: center;
+    background-color: ${color.stardust};
+    border-radius: ${radius.md};
+  }
+`
+
+export interface RadioWithLabelProps extends RadioProps {
+  label: string
+  ref?: React.Ref<HTMLInputElement>
+}
+
+export const RadioWithLabel: React.FC<RadioWithLabelProps> = forwardRef(
```

Comment:
**suggestion:** Having all variations in the same file is okay for small components, but might get complex for large ones. What do you think of having a separate file for each one of the variations and then export them all from the index?

https://trusz.github.io/posts/barrel-pattern-in-typescript/